packages/core/src/shareback/create-and-upload-extrinsic-object.ts

```diff
@@ -0,0 +1,62 @@
+import { DocumentReference, Organization } from "@medplum/fhirtypes";
+import { S3Utils } from "../external/aws/s3";
+import { createExtrinsicObjectXml } from "../external/carequality/dq/create-metadata-xml";
+import { createPatientUniqueId } from "../external/carequality/shared";
+import { isOrganization } from "../external/fhir/shared";
+import { out } from "../util/log";
+
+const { log } = out("Core Create and Upload Extrinsic Object");
+
+export async function createAndUploadDocumentMetadataFile({
+  s3Utils,
+  cxId,
+  patientId,
+  docId,
+  size,
+  docRef,
+  organization,
+  metadataFileName,
+  destinationBucket,
+  mimeType,
+}: {
+  s3Utils: S3Utils;
+  cxId: string;
+  patientId: string;
+  docId: string;
+  size: string;
```

Comment:
I see at least of the callers is sending empty string as a workaround when its not available.

I suggest we make this function accept optional/undefined `size` and decides how to better deal with it.

It shouldn't be a responsibility of the callers/clients to tell what's the alternative value when a size can't be determined.